src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -886,28 +886,14 @@ private static boolean isAstInside(DetailAST tree, DetailAST ast) {
      * @param ident ident token.
      * @return true if field can be referenced from a static context.
      */
-    private boolean canBeReferencedFromStaticContext(DetailAST ident) {
-        AbstractFrame variableDeclarationFrame = findFrame(ident, false);
-        while (variableDeclarationFrame.getType() == FrameType.BLOCK_FRAME
-            || variableDeclarationFrame.getType() == FrameType.FOR_FRAME) {
-            variableDeclarationFrame = variableDeclarationFrame.getParent();
-        }
-
+    private static boolean canBeReferencedFromStaticContext(DetailAST ident) {
         boolean staticContext = false;
 
-        if (variableDeclarationFrame.getType() == FrameType.CLASS_FRAME) {
```

Comment:
This mutation is what cascades a bunch of the others.

The issue is the above while and frame loop is what keeps looking the top level class. That is what caused the NPE at https://github.com/checkstyle/checkstyle/issues/12629 when I fooled into thinking it found the top level class when it found the catch frame.

`getCodeBlockDefinitionToken` also does its own scan for the top method/static init.

We need `variableDeclarationFrame` to find a non-class frame that will give us a different static value then the `getCodeBlockDefinitionToken` will. I could not find anything even with weird code.

Reply:
Associated suppression:
```
  <mutation unstable="false">
    <sourceFile>RequireThisCheck.java</sourceFile>
    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck</mutatedClass>
    <mutatedMethod>canBeReferencedFromStaticContext</mutatedMethod>
    <mutator>org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_IF</mutator>
    <description>removed conditional - replaced equality check with true</description>
    <lineContent>if (variableDeclarationFrame.getType() == FrameType.CLASS_FRAME) {</lineContent>
  </mutation>

```